api/airtable.ts

```diff
@@ -9,38 +9,20 @@ export class AirtableClient {
 
   async getById<T>(
     tableName: string,
-    keys: (keyof T)[],
     id: string,
   ): Promise<T> {
     const record = await this.base.table(tableName).find(id)
-
-    const object: {[key: string]: any} = {}
-    keys.forEach(key => {
-      object[key.toString()] = record.get(key.toString())
-    })
-    return object as T
+    return record.fields as unknown as T
   }
 
-  async getAll<T>(
+  async getAll<T extends object>(
     tableName: string,
-    keys: (keyof T)[],
     filter: string = '',
   ): Promise<T[]> {
-    const objects: T[] = []
-
     const records = await this.base.table(tableName).select({
       filterByFormula: filter,
     }).all()
 
-    // Map records to object specfied by keys
-    records.forEach(record => {
-      const object: {[key: string]: any} = {}
-      keys.forEach(key => {
-        object[key.toString()] = record.get(key.toString())
-      })
-      objects.push(object as T)
-    })
-
-    return objects as T[]
+    return records.map(record => record.fields) as T[]
```

Comment:
There was no need to iterate through a list of keys to get the fields we needed from each Airtable record 🤦🏼

Reply:
lol